src/components/Base/Switch/index.jsx

```diff
@@ -0,0 +1,53 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+import styles from './index.scss';
+
+export default class Switch extends PureComponent {
+  static propTypes = {
+    type: PropTypes.string,
+    onText: PropTypes.string,
+    offText: PropTypes.string,
+    onChange: PropTypes.func,
+    disabled: PropTypes.bool,
+    checked: PropTypes.bool,
+  };
+
+  static defaultProps = {
+    type: 'primary',
+    onChange: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      on: props.checked,
+    };
+  }
+
+  toggleSwitch = () => {
+    const on = !this.state.on;
+    this.setState({ on }, this.props.onChange(on));
```

Comment:
use setState callback, need send message to parent component ?

Reply:
Because props can control components, all of them require a callback to notify the parent when the state changes.